### ISSUE TITLE
assertion failed: self.font_size.map_or(true, |f| f == font_size)

### ISSUE THREAD

**Issue by mateon1** [servo/servo#18545]
Found on http://meteofrance.com

```gdb
assertion failed: self.font_size.map_or(true, |f| f == font_size) (thread <unnamed>, at /shared/dev/rust/servo/components/style/rule_cache.rs:28)
stack backtrace:
   0:     0x557516055184 - backtrace::backtrace::libunwind::trace
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/libunwind.rs:53
                         - backtrace::backtrace::trace<closure>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/mod.rs:42
   1:     0x55751604f9af - backtrace::capture::{{impl}}::new
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/capture.rs:64
   2:     0x55750f5b3ab6 - servo::main::{{closure}}
                        at /shared/dev/rust/servo/ports/servo/main.rs:130
   3:     0x557516335146 - std::panicking::rust_panic_with_hook
                        at /checkout/src/libstd/panicking.rs:578
   4:     0x55751502ddd5 - std::panicking::begin_panic<&str>
                        at /checkout/src/libstd/panicking.rs:538
   5:     0x557514f34884 - style::rule_cache::{{impl}}::set_font_size_dependency
                        at /shared/dev/rust/servo/components/style/rule_cache.rs:28
   6:     0x557515364f88 - style::values::specified::length::{{impl}}::reference_font_size_and_length
                        at /shared/dev/rust/servo/components/style/values/specified/length.rs:136
   7:     0x557515364c7f - style::values::specified::length::{{impl}}::to_computed_value
                        at /shared/dev/rust/servo/components/style/values/specified/length.rs:109
   8:     0x55751516a914 - style::values::computed::length::{{impl}}::to_computed_value
                        at /shared/dev/rust/servo/components/style/values/computed/length.rs:34
   9:     0x55751516aa3a - style::values::computed::length::{{impl}}::to_computed_value
                        at /shared/dev/rust/servo/components/style/values/computed/length.rs:57
  10:     0x5575151abaaf - style::values::generics::text::{{impl}}::to_computed_value<style::values::specified::length::Length>
                        at /shared/dev/rust/servo/components/style/values/generics/text.rs:36
  11:     0x5575151d787c - style::properties::longhands::letter_spacing::cascade_property
                        at /shared/dev/rust/servo/target/debug/build/style-1a72b88f5e4bdec5/out/properties.rs:19866
  12:     0x557514fd3692 - style::properties::apply_declarations<closure,core::iter::FlatMap<style::rule_tree::SelfAndAncestors, core::iter::FilterMap<core::iter::Rev<style::properties::declaration_block::DeclarationImportanceIterator>, closure>, closure>>
                        at /shared/dev/rust/servo/target/debug/build/style-1a72b88f5e4bdec5/out/properties.rs:80124
  13:     0x557514fd1667 - style::properties::cascade
                        at /shared/dev/rust/servo/target/debug/build/style-1a72b88f5e4bdec5/out/properties.rs:79809
  14:     0x55750fdce128 - style::style_resolver::{{impl}}::cascade_style<layout_thread::dom_wrapper::ServoLayoutElement>
                        at /shared/dev/rust/servo/components/style/style_resolver.rs:581
  15:     0x55750fdcff16 - style::style_resolver::{{impl}}::cascade_style_and_visited<layout_thread::dom_wrapper::ServoLayoutElement>
                        at /shared/dev/rust/servo/components/style/style_resolver.rs:304
  16:     0x55750fdcf87d - style::style_resolver::{{impl}}::resolve_primary_style<layout_thread::dom_wrapper::ServoLayoutElement>
                        at /shared/dev/rust/servo/components/style/style_resolver.rs:186
  17:     0x55750fdcefba - style::style_resolver::{{impl}}::resolve_style<layout_thread::dom_wrapper::ServoLayoutElement>
                        at /shared/dev/rust/servo/components/style/style_resolver.rs:205
  18:     0x55750fdd00ee - style::style_resolver::{{impl}}::resolve_style_with_default_parents::{{closure}}<layout_thread::dom_wrapper::ServoLayoutElement>
                        at /shared/dev/rust/servo/components/style/style_resolver.rs:243
  19:     0x55750fe6dfd3 - style::style_resolver::with_default_parent_styles<layout_thread::dom_wrapper::ServoLayoutElement,closure,style::style_resolver::ResolvedElementStyles>
                        at /shared/dev/rust/servo/components/style/style_resolver.rs:118
  20:     0x55750fdd007a - style::style_resolver::{{impl}}::resolve_style_with_default_parents<layout_thread::dom_wrapper::ServoLayoutElement>
                        at /shared/dev/rust/servo/components/style/style_resolver.rs:242
  21:     0x55751012f2c7 - style::traversal::compute_style<layout_thread::dom_wrapper::ServoLayoutElement>
                        at /shared/dev/rust/servo/components/style/traversal.rs:678
  22:     0x557510218f69 - style::traversal::recalc_style_at<layout_thread::dom_wrapper::ServoLayoutElement,layout::traversal::RecalcStyleAndConstructFlows,closure>
                        at /shared/dev/rust/servo/components/style/traversal.rs:477
  23:     0x5575101abf0c - layout::traversal::{{impl}}::process_preorder<layout_thread::dom_wrapper::ServoLayoutElement,closure>
                        at /shared/dev/rust/servo/components/layout/traversal.rs:66
  24:     0x55750ffc856a - style::parallel::top_down_dom<layout_thread::dom_wrapper::ServoLayoutElement,layout::traversal::RecalcStyleAndConstructFlows>
                        at /shared/dev/rust/servo/components/style/parallel.rs:191
                         - style::parallel::traverse_nodes::{{closure}}<layout_thread::dom_wrapper::ServoLayoutElement,layout::traversal::RecalcStyleAndConstructFlows,smallvec::Drain<style::dom::SendNode<layout_thread::dom_wrapper::ServoLayoutNode>>>
                        at /shared/dev/rust/servo/components/style/parallel.rs:285
  25:     0x55750ff197fd - rayon_core::scope::{{impl}}::execute_job_closure::{{closure}}<closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/scope/mod.rs:354
  26:     0x55750ff1125a - std::panic::{{impl}}::call_once<(),closure>
                        at /checkout/src/libstd/panic.rs:296
  27:     0x557510092bb2 - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panicking.rs:480
  28:     0x55751633c2dc - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  29:     0x557510091dac - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /checkout/src/libstd/panicking.rs:459
  30:     0x55750ff612f5 - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panic.rs:361
  31:     0x55750ff1a104 - rayon_core::unwind::halt_unwinding<closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/unwind.rs:19
  32:     0x55750fdd1e8a - rayon_core::scope::{{impl}}::execute_job_closure<closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/scope/mod.rs:354
  33:     0x55750fdd1250 - rayon_core::scope::{{impl}}::execute_job<closure>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/scope/mod.rs:343
  34:     0x55750fdd2e67 - rayon_core::scope::{{impl}}::spawn::{{closure}}<closure>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/scope/mod.rs:274
  35:     0x55750fe96553 - rayon_core::job::{{impl}}::execute<closure>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/job.rs:142
  36:     0x557515c79c4f - rayon_core::job::{{impl}}::execute
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/job.rs:55
  37:     0x557515c7c1ee - rayon_core::registry::{{impl}}::execute
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:476
  38:     0x557515c7bb7f - rayon_core::registry::{{impl}}::wait_until_cold<rayon_core::latch::CountLatch>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:460
  39:     0x557515c7273f - rayon_core::registry::{{impl}}::wait_until<rayon_core::latch::CountLatch>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:436
  40:     0x557515c72d2c - rayon_core::scope::{{impl}}::steal_till_jobs_complete
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/scope/mod.rs:382
  41:     0x55750ff199a5 - rayon_core::scope::scope::{{closure}}<closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/scope/mod.rs:257
  42:     0x55750ff1a6f0 - rayon_core::registry::in_worker<closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:597
  43:     0x55750fdd3015 - rayon_core::scope::scope<closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/scope/mod.rs:248
  44:     0x55750ffc27c9 - style::driver::traverse_dom::{{closure}}<layout_thread::dom_wrapper::ServoLayoutElement,layout::traversal::RecalcStyleAndConstructFlows>
                        at /shared/dev/rust/servo/components/style/driver.rs:92
  45:     0x55750fffcb0d - rayon_core::job::{{impl}}::execute::{{closure}}<rayon_core::latch::LockLatch,closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/job.rs:99
  46:     0x55750ff11017 - std::panic::{{impl}}::call_once<(),closure>
                        at /checkout/src/libstd/panic.rs:296
  47:     0x557510092a4f - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panicking.rs:480
  48:     0x55751633c2dc - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  49:     0x55751009270c - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /checkout/src/libstd/panicking.rs:459
  50:     0x55750ff61152 - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panic.rs:361
  51:     0x55750ff19b01 - rayon_core::unwind::halt_unwinding<closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/unwind.rs:19
  52:     0x55750feb9173 - rayon_core::job::{{impl}}::execute<rayon_core::latch::LockLatch,closure,()>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/job.rs:99
  53:     0x557515c79c4f - rayon_core::job::{{impl}}::execute
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/job.rs:55
  54:     0x557515c7c1ee - rayon_core::registry::{{impl}}::execute
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:476
  55:     0x557515c7bb7f - rayon_core::registry::{{impl}}::wait_until_cold<rayon_core::latch::CountLatch>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:460
  56:     0x557515c7b99f - rayon_core::registry::{{impl}}::wait_until<rayon_core::latch::CountLatch>
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:436
  57:     0x557515c7ca10 - rayon_core::registry::main_loop
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:559
  58:     0x557515c66e56 - rayon_core::registry::{{impl}}::new::{{closure}}
                        at /shared/dev/rust/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.2.0/src/registry.rs:145
  59:     0x557515c5c1f2 - std::sys_common::backtrace::__rust_begin_short_backtrace<closure,()>
                        at /checkout/src/libstd/sys_common/backtrace.rs:136
  60:     0x557515c740ca - std::thread::{{impl}}::spawn::{{closure}}::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:394
  61:     0x557515c73132 - std::panic::{{impl}}::call_once<(),closure>
                        at /checkout/src/libstd/panic.rs:296
  62:     0x557515c5d734 - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panicking.rs:480
  63:     0x55751633c2dc - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  64:     0x557515c5d515 - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /checkout/src/libstd/panicking.rs:459
  65:     0x557515c73fd9 - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panic.rs:361
  66:     0x557515c5d0ee - std::thread::{{impl}}::spawn::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:393
  67:     0x557515c64447 - alloc::boxed::{{impl}}::call_box<(),closure>
                        at /checkout/src/liballoc/boxed.rs:682
  68:     0x557516333dcb - alloc::boxed::{{impl}}::call_once<(),()>
                        at /checkout/src/liballoc/boxed.rs:692
                         - std::sys_common::thread::start_thread
                        at /checkout/src/libstd/sys_common/thread.rs:21
                         - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /checkout/src/libstd/sys/unix/thread.rs:84
  69:     0x7f95eb628493 - start_thread
  70:     0x7f95eb155abe - __clone
  71:                0x0 - <unknown>
ERROR:servo: assertion failed: self.font_size.map_or(true, |f| f == font_size)
```

**Comment by mateon1** [comment]
Crashing testcase:
```html
<p style="font-size: calc(1em + 1px)"></p>
```

**Comment by emilio** [comment]
Should be fixed in #18538, please shout out if not and I'll reopen it. Thanks!
